android/buildAndroidBOINC-CI.sh

```diff
@@ -129,18 +134,31 @@ fi
 
 export COMPILEOPENSSL="no"
 export COMPILECURL="no"
-export NDK_FLAGFILE="$PREFIX/NDK-${NDK_VERSION}-${arch}_done"
+export NDK_FLAGFILE="$PREFIX/NDK-${NDK_VERSION}_done"
 export CURL_FLAGFILE="$PREFIX/curl-${CURL_VERSION}-${arch}_done"
 export OPENSSL_FLAGFILE="$PREFIX/openssl-${OPENSSL_VERSION}-${arch}_done"
+export CREATE_NDK_FOLDER=${CREATE_NDK_FOLDER:-"no"}
+
+createNDKFolder()
+{
+    if [ $CREATE_NDK_FOLDER = "no" ]; then
+        rm -rf "$BUILD_DIR/android-ndk-r${NDK_VERSION}"
+        wget -c --no-verbose -O /tmp/ndk.zip https://dl.google.com/android/repository/android-ndk-r${NDK_VERSION}-linux-x86_64.zip
+        unzip -qq /tmp/ndk.zip -d $BUILD_DIR
+        export CREATE_NDK_FOLDER="yes"
+    fi
+}
 
-if [ ! -e "${NDK_FLAGFILE}" ]; then
-    rm -rf "$BUILD_DIR/android-ndk-r${NDK_VERSION}"
-    rm -rf "${PREFIX}/${arch}"
-    rm -f "${CURL_FLAGFILE}" "${OPENSSL_FLAGFILE}"
-    wget -c --no-verbose -O /tmp/ndk.zip https://dl.google.com/android/repository/android-ndk-r${NDK_VERSION}-linux-x86_64.zip
-    unzip -qq /tmp/ndk.zip -d $BUILD_DIR
-    touch "${NDK_FLAGFILE}"
+if [ ci = "yes" ]; then
+    createNDKFolder
```

Comment:
Looks like every time on CI cache folder will be deleted before each run

Reply:
No. The whole run in the CI it will create only once.
Inside the function there is a check that insure it.

Reply:
What check? `CREATE_NDK_FOLDER` ?
If yes - then between two runs (e.g. two commits) this folder will be rectreated again. Because on every run it will check that it is on ci, `CREATE_NDK_FOLDER` by default is `no` because no information about the state of this variable was saved from the previous run, and cache will be deleted

Reply:
This is not create the standalone tool like it was before. 
Before the change it was downloading the all ndk folder very run just for compile boinc.  
After the change it do the same. It not cache the tool chain, because there isn't standalone tool chain to cache like before.
It use the ndk folder to  compile without create any tool chain standalone tool.

Reply:
Every run it download the ndk folder once. It don't save it between runs.

Reply:
And every run we will rebuild openssl and curl.
This need to be fixed in some way

Reply:
No, we don't need to rebuild openssl and curl because it save in the cache. 
It just download the ndk folder for boinc compilation, compile boinc with openssl and curl from the cache.

Reply:
In the next version of the ndk, when they reduce the size of the ndk zip file lower then 1G , I will cache the ndk folder too.